File: app/src/main/java/ly/pp/justpiano3/entity/WaterfallNote.java
package ly.pp.justpiano3.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * 瀑布流音符
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@Accessors(chain = true)
public class WaterfallNote {

    /**
     * 绘制区域左边界坐标
     */
    private float left;

    /**
     * 绘制区域右边界坐标
     */
    private float right;

    /**
     * 音符开始时间，以曲谱开始时间为0开始累计计算的值，毫秒
     */
    private int startTime;

    /**
     * 音符开始时间，以曲谱开始时间为0开始累计计算的值，毫秒
     */
    private int endTime;

    /**
     * 是否为左手
     */
    private boolean leftHand;

    /**
     * 音高
     */
    private int pitch;

    /**
     * 音量
     */
    private int volume;

    /**
     * 音符持续时间计算
     */
    public int interval() {
        return endTime - startTime;
    }
}

File: app/src/main/java/ly/pp/justpiano3/entity/WaterfallNote.kt
package ly.pp.justpiano3.entity

/**
 * 瀑布流音符
 */
data class WaterfallNote(
    /**
     * 绘制区域左边界坐标
     */
    var left: Float = 0f,

    /**
     * 绘制区域右边界坐标
     */
    var right: Float = 0f,

    /**
     * 绘制区域上边界坐标，以曲谱开始时间为0开始累计计算的坐标值
     */
    var top: Float = 0f,

    /**
     * 绘制区域下边界坐标，以曲谱开始时间为0开始累计计算的坐标值
     */
    var bottom: Float = 0f,

    /**
     * 瀑布流音块颜色
     */
    var color: Int = 0,

    /**
     * 音高
     */
    var pitch: Byte = 0,

    /**
     * 音量
     */
    var volume: Byte = 0,
)